File: Applied Course/4.Problem Solving/1.Problems on Array/8.Replace Elements with Greatest Element on Right Side.py
# https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/
# Logic-1
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        l=[]
        for i in range(len(arr)-1):
            l.append(max(arr[i+1:]))
        l.append(-1)
        return l
# Logic-2
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n=len(arr)
        max_ele_seen_so_far=arr[n-1]
        arr[n-1]=-1
        for i in range(n-2,-1,-1):
            temp=arr[i]
            arr[i]=max_ele_seen_so_far
            if temp>max_ele_seen_so_far:
                max_ele_seen_so_far=temp
        return arr

File: Applied Course/4.Problem Solving/1.Problems on Array/8.Replace Elements with Greatest Element on Right Side.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n=arr.size();
        int max_ele=arr[n-1];
        arr[n-1]=-1;
        for(int i=n-2;i>=0;i--)
        {
            int temp=arr[i];
            arr[i]=max_ele;
            if(temp>max_ele)
                max_ele=temp;
        }
        return arr;
    }
};
int main()
{
    // https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/
    return 0;
}